homeassistant/components/alexa/handlers.py

```diff
@@ -520,21 +521,49 @@
     """Process an adjust volume step request."""
     # media_player volume up/down service does not support specifying steps
     # each component handles it differently e.g. via config.
-    # For now we use the volumeSteps returned to figure out if we
-    # should step up/down
-    volume_step = directive.payload["volumeSteps"]
+    # This workaround will simply call the volume up/Volume down the amount of steps asked for
+    # When no steps are called in the request, Alexa sends a default of 10 steps which for most
+    # purposes is too high.  A device_state_attribute can be set in the media player to set this to a new default
+    # The default without the attribute is set 1 in this case.
     entity = directive.entity
-
-    data = {ATTR_ENTITY_ID: entity.entity_id}
-
-    if volume_step > 0:
-        await hass.services.async_call(
-            entity.domain, SERVICE_VOLUME_UP, data, blocking=False, context=context
-        )
-    elif volume_step < 0:
-        await hass.services.async_call(
-            entity.domain, SERVICE_VOLUME_DOWN, data, blocking=False, context=context
-        )
+    volume_int = int(directive.payload["volumeSteps"])
+    is_default = bool(directive.payload["volumeStepsDefault"])
+    default_steps = 1
+    sleep_delay = 0.5
+
+    if "volume_steps_default" in entity.attributes:
+        try:
+            default_steps = int(entity.attributes["volume_steps_default"])
+        except ValueError:
+            default_steps = 1
+
+    if is_default:
+        if volume_int < 0:
+            volume_int = -default_steps
+        else:
+            volume_int = default_steps
+
+    if volume_int != 0:
+        data = {ATTR_ENTITY_ID: entity.entity_id}
+
+        for _ in range(0, abs(volume_int)):
+            if volume_int > 0:
+                await hass.services.async_call(
+                    entity.domain,
+                    SERVICE_VOLUME_UP,
+                    data,
+                    blocking=False,
+                    context=context,
+                )
+            elif volume_int < 0:
+                await hass.services.async_call(
+                    entity.domain,
+                    SERVICE_VOLUME_DOWN,
+                    data,
+                    blocking=False,
+                    context=context,
+                )
+            await asyncio.sleep(sleep_delay)
```

Comment:
We shouldn't sleep. It's up to the receiving integration to throttle when necessary.